tests/frontend/test_decompositions.py

```diff
@@ -213,6 +213,72 @@ def test_random_unitary_phase_end(self, tol):
         assert np.allclose(U, qrec, atol=tol, rtol=0)
 
 
+class TestRectangularSymmetricDecomposition:
+    """Tests for linear interferometer decomposition into rectangular grid of
+    phase-shifters and pairs of symmetric beamsplitters"""
+
+    def test_unitary_validation(self):
+        """Test that an exception is raised if not unitary"""
+        A = np.random.random([5, 5]) + 1j * np.random.random([5, 5])
+        with pytest.raises(ValueError, match="matrix is not unitary"):
+            dec.rectangular_symmetric(A)
+
+    @classmethod
+    def S(cls, m, n, internal_phase, external_phase, nmax):
+        r"""a section composed by external phase shift, BS, internal phase shift, BS
+
+        .. todo::
+
+           Move this function into decompositions and maybe rename and / or optimize.
```

Comment:
:+1:

Reply:
you have a name suggestion? Maybe 'MachZehnder'?

Reply:
I like that name, it seems to be the term I hear most often (assuming it is accurate and won't mislead people, of course)

Reply:
How about `mach_ zehnder` to match the python convention

Reply:
I would much prefer to not use names. Symmetric is quite descriptive.

Reply:
@nquesada, the question is whether this test function, which essentially is a modified BSGate with the rectangular_symmetric phase convention (i.e. taking two phases and outputting a matrix), should go into decompositions.py, akin to the function T that represents a BSGate in the Clements decomposition. If it does, it needs a name. Currently I randomly chose S, but it could also be mach_zehner, or even MZGate. What option would you suggest here (S, mach_zehnder, MZGate, not moving into decompositions.py at all, or something else)?

Reply:
I chose the option `mach_zehnder` and added a commit for this now. But I can undo it if you want.

Reply:
> the question is whether this test function, which essentially is a modified BSGate with the rectangular_symmetric phase convention (i.e. taking two phases and outputting a matrix), should go into decompositions.py

IMO it should definitely go into `decompositions.py` - it is useful to have the decomposition separate from Strawberry Fields classes and objects.

In terms of accessing this operation from the SF frontend, this should be done in a future PR, but I can think of two options:

* Introduce an `MZgate(theta, phi)` operation in Strawberry Fields that uses the above decomposition, compiling to a combination of `Rgate`'s and `BSgate(pi/4, pi/2)`.

* Add a keyword argument to `Interferometer` allowing you to change the geometry of the unitary decomposition, i.e.,

  ```python
  Interferometer(U, geometry='rectangular')
  Interferometer(U, geometry='triangular')
  Interferometer(U, geometry='mach_zehnder') # or maybe 'hardware'?
  ```

As both have valid use cases, I'm leaning towards implementing both approaches.